File: 14_LongestCommonPrefix/longestCommonPrefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs: return ''
        s_min, s_max = min(strs), max(strs)
        for i, c in enumerate(s_min):
            if c != s_max[i]:
                return s_min[:i]
        return s_min

File: 14_LongestCommonPrefix/longestCommonPrefix.go
func longestCommonPrefix(strs []string) string {
    if len(strs) == 0 {
        return ""
    }

    for i := 0; i < len(strs[0]); i++ {
        for _, s := range strs {
            if i >= len(s) || strs[0][i] != s[i] {
                return strs[0][:i]
            }
        }
    }

    return strs[0]
}
